src/xml.ts

```diff
@@ -115,10 +115,10 @@ function getAllTests(task: Task): Test[] {
 }
 
 function getDurationAttribute(test: Test): string {
-    const duration = test.result?.duration;
+    const duration = test.result?.duration ?? test.file?.result?.duration;
 
-    if (typeof duration !== 'number') {
-        return '';
+    if (typeof duration !== "number") {
+        return ` duration="0"`;
```

Comment:
This seems correct. Duration is a mandatory field based on [SonarQube documentation](https://docs.sonarqube.org/latest/analysis/generic-test/#header-2).

> Inside a file element, insert a testCase for each test run by unit tests. It can have the following attributes/children:
>
> - testCase (mandatory)
>
>   - duration (mandatory): long value in milliseconds